Update mocking-localizer.md (#1260)
bUnit-dev/bUnit: docs/site/docs/test-doubles/mocking-localizer.md

title: Mocking Localization via IStringLocalizer
---

> <p>This is very simple. First in your setup add the following:</p>

Author: I changed it.

> <code>TestContext.Services.AddLocalization();</code>

Reviewer: Our "pattern" is to have whole blogs of code like
&#96;&#96;&#96;csharp
TestContext.Services.AddLocalization();
&#96;&#96;&#96;

But I would go a step further:
A minimal example of the production code enriched by the test-code would be ideal. But that is my opinion. Let's hear if @egil has a different stance here.

Author: I did as you suggested. It doesn't show as code when I view it, but I'm assuming that something processes it when pushed up.

title: Mocking Localization via IStringLocalizer
---

> <p>There are just two steps. First in your setup add the following:</p>

Reviewer: Another small amendment:
Bunit documentation "style" doesn't use "you" or "your".
Just things like:
```suggestion
<p>There are just two steps. First in the setup add the following:</p>
```

And so on.

```csharp
TestContext.Services.AddLocalization();
```

> <p>Then in your test code, when you need the localized string to compare, you write the following:</p>

Reviewer: ```suggestion
In the test code, when there is a need for the localized string to compare, the following is written:
```

```csharp
var localizer = ctx.Services.GetService<IStringLocalizer<SharedStrings>>();

>   ```

Reviewer: Some smaller whitespaces, that can be removed

Author: please make any changes you want.

> <p>Where SharedStrings.cs (you can name this anything you want) that has the resource files such as `SharedStrings.en.resx`</p>

Reviewer: ```suggestion
Where `SharedStrings.cs` (which may be named as desired) contains the resource files such as `SharedStrings.en.resx`.
```
